pkg/api/customization/node/node.go

```diff
@@ -48,16 +48,35 @@ func Formatter(apiContext *types.APIContext, resource *types.RawResource) {
 	if nodeTemplateID == nil && customConfig == nil {
 		delete(resource.Links, "remove")
 	}
-	if resource.Values[client.NodeFieldWorker] != true ||
-		resource.Values["state"] == "registering" ||
-		resource.Values["state"] == "unavailable" {
+	if resource.Values[client.NodeFieldWorker] != true {
 		return
 	}
-	if resource.Values["state"] == "draining" {
+	state := resource.Values["state"]
+	allowed := state == "active" || state == "cordoned" || state == "draining" || state == "drained"
+	if !allowed {
+		return
+	}
+	if state == "draining" {
+		conditions := convert.ToMapSlice(resource.Values["conditions"])
+		for _, cond := range conditions {
+			if cond["type"] == "Drained" && cond["status"] == "False" {
+				msg := convert.ToString(cond["message"])
+				if strings.Contains(msg, "--ignore-daemonsets") ||
```

Comment:
can we put all the messages to the map or slice, and rework this loop to be more generic?